app/controllers/spree/admin/orders_controller_decorator.rb

```diff
@@ -27,6 +27,23 @@ def index
     # within the page then fetches the data it needs from Api::OrdersController
   end
 
+  # Re-implement spree method so that it redirects to edit instead of rendering edit
+  #   This allows page reloads while adding variants to the order (/edit), without being redirected to customer details page (/update)
+  def update
+    unless @order.update_attributes(params[:order]) && @order.line_items.present?
+      @order.errors.add(:line_items, Spree.t('errors.messages.blank')) if @order.line_items.empty?
```

Comment:
Isn't this implemented in the model layer? If not, it should be there and be consistent with the other places from where we can place an order (the shopfront).

Reply:
I think this is specific to the BO
see the spree code:
https://github.com/spree/spree/blob/58ef94cdb17310d62e110dd92a7d4b4d1f270335/backend/app/controllers/spree/admin/orders_controller.rb#L82

the orders in the shopfront are the carts so they don't have line items and the presence of line items is not relevant most of the time.
what you suggest would imply validating the present of line items in a specific stage of the order workflow.
I wouldnt go that way... what do you think?

Reply:
In that case is probably is wiser to stick to this :+1: